sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/implementation/notifications/fcm/PushNotificationClient.java

```diff
@@ -284,4 +310,42 @@ public void run() {
         this.registrationRenewScheduleTimer = new Timer("PushNotificationRegistrarRenewTimer");
         this.registrationRenewScheduleTimer.schedule(task, delayMs);
     }
+
+    private void refreshEncryptionKeys() throws Throwable {
+        if (this.keyGenerator == null) {
+            this.keyGenerator = KeyGenerator.getInstance("AES");
+            this.keyGenerator.init(KEY_SIZE);
+        }
+
+        this.previousCryptoKey = this.cryptoKey;
+        this.previousAuthKey = this.authKey;
+
+        this.cryptoKey = this.keyGenerator.generateKey();
+        this.authKey = this.keyGenerator.generateKey();
+    }
+
+    private String decryptPayload(String encryptedPayload) throws Throwable {
+        this.logger.info(" Decrypting payload.");
```

Comment:
Is there a particular reason we add a space at the beginning of these logs?

Reply:
The reason is if the class name is too long, in log this.logger.info() will be printed directly after class name without a space. So I manually added a space here to separate the class name & log messages...